File: javascript/programmers/javascript/level2/피보나치수.js
const n = 3;

const solution = (n) => {
  let prev1 = 0,
    prev2 = 1;
  let val = 0;
  for (let i = 2; i <= n; i++) {
    val = prev1 + prev2;
    prev1 = prev2 % 1234567;
    prev2 = val % 1234567;
  }
  return val % 1234567;
};

console.log(solution(n));

File: python/programmers/level2/피보나치수.py
def solution(n):
    mod = 1234567

    pibo = [0] * (n + 1)
    pibo[1] = 1
    for i in range(2, n + 1):
        pibo[i] = (pibo[i - 1] + pibo[i - 2]) % mod
    return pibo[n] % mod


print(solution(3))
